frontend/userpatch.lua

```diff
@@ -55,16 +56,7 @@ local function runUserPatchTasks(dir, priority)
         return -- nothing to do
     end
 
-    local function addLeadingZeroes(d)
-        local dec, n = string.match(d, "(%.?)0*(.+)")
-        return #dec > 0 and ("%.12f"):format(d) or ("%s%03d%s"):format(dec, #n, n)
-    end
-    local function sorting(a, b)
-        return tostring(a):gsub("%.?%d+", addLeadingZeroes)..("%3d"):format(#b)
-            < tostring(b):gsub("%.?%d+", addLeadingZeroes)..("%3d"):format(#a)
-    end
-
-    table.sort(patches, sorting)
+    table.sort(patches, sort.natsort)
```

Comment:
And because these natural transformed strings has no need to stay in memory, you probably should use a tag:
```lua
sort.natsort_set_cache("userpatches")
table.sort(patches, sort.natsort)
sort.natsort_clear_cache("userpatches")
```